osu.Game/Rulesets/Difficulty/DifficultyAttributes.cs

```diff
@@ -24,5 +32,11 @@ public DifficultyAttributes(Mod[] mods, Skill[] skills, double starRating)
             Skills = skills;
             StarRating = starRating;
         }
+
+        public virtual IEnumerable<(int attributeId, object value)> ToDatabaseAttributes() => Enumerable.Empty<(int, object)>();
```

Comment:
PR seems fine, but might be a good opportunity to xmldoc at least this base class?